packages/db-objects-redis/lib/objects/objectsInRedisClient.js

```diff
@@ -390,21 +392,45 @@ class ObjectsInRedisClient {
                 });
             }
 
-            this.log.debug(this.namespace + ' Objects client initialize lua scripts');
+            if (!this.client) {
+                return;
+            }
+
+            // for controller v4 we have to check if we can use the new lua scripts and set logic
+            // TODO: remove this backward shim if controller v4.0 is old enough
+            const keys = await this._getKeysViaScan(`${this.objNamespace}system.host.*`);
+
+            this.useSets = true;
+
+            try {
+                for (const key of keys) {
+                    let obj = await this.client.get(key);
+                    obj = JSON.parse(obj);
+                    if (obj && obj.type === 'host' && obj.common && obj.common.installedVersion &&
+                        semver.lt(obj.common.installedVersion, '4.0.0')) {
+                        // one of the host has a version smaller 4, we have to use legacy db
+                        this.useSets = false;
+                        this.log.info('Sets unsupported');
+                    }
+                }
+            } catch (e) {
+                this.log.error(`Cannot determine Lua scripts strategy: ${e.message}`);
+                return;
```

Comment:
then use v3 for now?

Reply:
problem here is if one can determine v4 and the other not -> incompatible, so my propose was to not start